identity/src/iosMain/kotlin/com/android/identity/nfc/scanNfcTag.ios.kt

```diff
@@ -0,0 +1,139 @@
+package com.android.identity.nfc
+
+import com.android.identity.util.Logger
+import com.android.identity.util.toKotlinError
+import kotlinx.coroutines.CancellableContinuation
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.IO
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+import platform.CoreNFC.NFCISO7816TagProtocol
+import platform.CoreNFC.NFCPollingISO14443
+import platform.CoreNFC.NFCTagProtocol
+import platform.CoreNFC.NFCTagReaderSession
+import platform.CoreNFC.NFCTagReaderSessionDelegateProtocol
+import platform.Foundation.NSError
+import platform.CoreNFC.NFCErrorDomain
+import platform.CoreNFC.NFCReaderSessionInvalidationErrorUserCanceled
+import platform.darwin.NSObject
+import kotlin.coroutines.resumeWithException
+
+private class NfcTagReader<T> {
+
+    private class DialogCanceledException : Throwable()
+
+    companion object {
+        private const val TAG = "NfcTagReader"
+    }
+
+    val session: NFCTagReaderSession
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val tagReaderSessionDelegate = object : NSObject(), NFCTagReaderSessionDelegateProtocol {
+        override fun tagReaderSession(
+            session: NFCTagReaderSession,
+            didInvalidateWithError: NSError
+        ) {
+            val nsError = didInvalidateWithError
+            if (nsError.domain == NFCErrorDomain &&
+                nsError.code == NFCReaderSessionInvalidationErrorUserCanceled) {
+                continuation?.resumeWithException(DialogCanceledException())
+                continuation = null
+            } else {
+                continuation?.resumeWithException(nsError.toKotlinError())
+                continuation = null
+            }
+        }
+
+        override fun tagReaderSessionDidBecomeActive(
+            session: NFCTagReaderSession
+        ) {
+        }
+
+        override fun tagReaderSession(
+            session: NFCTagReaderSession,
+            didDetectTags: List<*>
+        ) {
+            val tags = didDetectTags as List<NFCTagProtocol>
+            val tag = tags[0]
```

Comment:
Is it safe to ignore all but the first tag? Are there any edge cases to worry about here?
Maybe add a comment about why this is safe (assuming it is).

Reply:
I think so. Although I do think it's possible several tags can be in the field if using e.g. a wallet strapped on the back on your phone. That way your credit cards are always in the field and I imagine you need to look at all the tags. I'll investigate, will leave a TODO in the code so we don't forget.